### PR TITLE
feat: add security.default_task and openshift host options to helm chart [DPS-204]

### PR DESCRIPTION
**Pull request by asnelling**
## Description

add `security.default_task` and openshift host options to helm chart [DPS-204]

## Test Plan

Install Determined using helm chart as in [this guide](https://docs.determined.ai/latest/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-k8s/install-on-kubernetes.html):
- without changing `values.yaml`
- with `security.defaultTask` configured in `values.yaml` as:
   ```yaml
   security:
     defaultTask:
       user: det-nobody
       uid: 65533
       group: det-nobody
       gid: 65533
   ```
   Verify changes propagated to `master.yaml`:
   ```Shell
   $ det -u admin master config | grep -e det-nobody -e 65533
       gid: 65533
       group: det-nobody
       uid: 65533
       user det-nobody
   ```

## Checklist
- [x] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.
- [x] If modifying `/webui/react/src/shared/` verify `make -C webui/react test-shared` passes.

[DPS-204]: https://determinedai.atlassian.net/browse/DPS-204?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: helm/charts/determined/templates/master-deployment.yaml [hunk 1]
```diff
@@ -40,11 +40,13 @@ spec:
...
             mountPath: /etc/determined/
             readOnly: true
-          {{- if .Values.tlsSecret }}
+          {{- if .Values.security }}
+          {{- if .Values.security.tlsSecret }}
           - name: tls-secret
             mountPath: {{ include "determined.secretPath" . }}
             readOnly: true
           {{ end }}
+          {{ end }}
```
Comment (dannysauer) [inline helm/charts/determined/templates/master-deployment.yaml:49]: We're oddly inconsistent with `{{ end }}`, `{{- end }}`, and `{{ end -}}` i this chart.  Maybe not really something that belongs in this PR, but probably something worth cleaning up.

File: helm/charts/determined/values.yaml [hunk 2]
```diff
@@ -22,14 +22,26 @@ useNodePortForMaster: false
...
+
+  # defaultTask sets the user and group that tasks will run as. For convenience, the default Determined
+  # environments contain an unprivileged user named det-nobody, which does have a writable HOME
+  # directory. The det-nobody user is a suitable default user when using the default Determined
+  # environment images and when running containers as root is not desired.
+  # defaultTask:
+  #   user: det-nobody
+  #   uid: 65533
+  #   group: det-nobody
+  #   gid: 65533
```
Comment (dannysauer) [inline helm/charts/determined/values.yaml:44]: I forget - is there a good reason to not use `-3` here so it'll overflow properly on both systems with 32-bit IDs and older systems with 16-bit IDs?  Is `65533` hard-coded elsewhere in the containers?

**Review by dannysauer — APPROVED** [review]
I don't see any technical issues, though I did point out a couple of _potential_ nits.

It also would likely be a good idea to at least include a changelog entry indicating that the hierarchical location of the `tlsSecret` parameter has changed.  I'm not sure how we typically do that.  I'm marking approved anyway; I'll let you figure out if those things need further changes or not. :)

File: helm/charts/determined/templates/NOTES.txt [hunk 3]
```diff
@@ -19,13 +19,15 @@ kubectl get service determined-master-service-{{ .Release.Name }} -n {{ .Release
 {{ end -}}
 
 {{ $httpOrHttps := "http" }}
```
Comment (varlogtim) [inline helm/charts/determined/templates/NOTES.txt:21]: Per the thread, if they make no changes and don't read the docs, this silently fails to set their secret. Include an error message if they still have `.Values.tlsSecret` set.

**Review by varlogtim — CHANGES_REQUESTED** [review]
Per slack thread, please handle the backwards compatibility case. Preferably by printing a warning/error message indicating that the location of the `tlsSecret` has moved.

File: helm/charts/determined/values.yaml (hunk 2 above)
Comment (varlogtim) [reply helm/charts/determined/values.yaml:44]: 65533 is hardcoded in our containers as `det-nobody`, I believe.

File: helm/charts/determined/templates/NOTES.txt (hunk 3 above)
Comment (dannysauer) [reply helm/charts/determined/templates/NOTES.txt:21]: I was looking for an error function, but of course, Helm (and go templates) are never what anyone really wants. :D

My current best idea is to use [required](https://helm.sh/docs/howto/charts_tips_and_tricks/#using-the-required-function) in a random template file to depend on the new value if the old is set; maybe something along the lines of this untested blob of horror:
```plain
{{- if .Values.tlsSecret }} 
{{- required "tlsSecret has moved to security.tslSecret" .Values.security.tlsSecret }}
{{- end }}
```
I don't know for sure if that works in a conditional, though, so :shrug:

Comment (dannysauer) [reply helm/charts/determined/templates/NOTES.txt:21]: Oh, sprig has a fail function.  So maybe `{{- fail "tlsSecret has moved to security.tslSecret" }}` would be a similar valid option to always fail if the old value is provided  I kinda like continuing to work if the new one is also added.  Dealer's choice, IMO.

Comment (asnelling) [reply helm/charts/determined/templates/NOTES.txt]: The decision was made to move `tlsSecret` back to it's original home and avoid the situation altogether.

**Review by varlogtim — APPROVED** [review]

**Review by dannysauer — APPROVED** [review]
This LGTM

**Comment by dannysauer** [comment]
Also, I'd say just go ahead and admin merge this. You could rebase and then run the teats again to get the python lint error to pass, but since there's no python change anyway, that seems pointless.

**Pull request merged**